File: DSA/C++/bubble_sort.cpp
// Description : Implementation of Bubble Sort algorithm in C++
#include <iostream>
using namespace std;

int main()
{
    int n;
    cout << "Enter number of elements: ";
    cin >> n;
    int *arr = new int[n];

    cout << "Enter " << n << " elements:" << endl;
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }

    // Bubble Sort Algorithm
    for (int i = 0; i < n - 1; i++)
    {
        for (int j = 0; j < n - i - 1; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                // Swap arr[j] and arr[j+1]
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }

    cout << "Sorted array: ";
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
    cout << endl;

    return 0;
}

File: DSA/Python/bubble_sort.py
# Description : Implementation of Bubble Sort algorithm in Python

def main():
    n = int(input("Enter number of elements: "))
    arr = []

    print(f"Enter {n} elements:")
    for _ in range(n):
        element = int(input())
        arr.append(element)

    for i in range(n - 1):
        for j in range(n - i - 1):
            if arr[j] > arr[j + 1]:
                # Pythonic swap
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

    print("Sorted array: ", end="")
    # Print elements on the same line
    for i in range(n):
        print(arr[i], end=" ")
    print()


if __name__ == "__main__":
    main()
